### ISSUE TITLE
facet_highlight()

### ISSUE THREAD

**Issue by yutannihilation** [yutannihilation/gghighlight#14]

**Comment by wilfredom** [comment]
Heya—thanks for the awesome package. Is there a timeline for this feature? It would really cinch the gold 🥇 !

**Comment by yutannihilation** [comment]
> Is there a timeline for this feature?

No, I don't even remember what I was to implement... :P 

BTW, what do you expect for `facet_highlight()`? Is that just a shortcut for `gghighlight(...) + facet_foo(...)`? I want to know your use cases.

**Comment by wilfredom** [comment]
Lol, fair enough. The use case is that gghighlight doesn't "respect" facetting: the non-highlighted geoms come from the entire data set, rather than each facet. 

That is, imagine facet_var has N values, and I would like to highlight based on another variable within each facet. Right now, instead of the two commands you described above, I have to create each facet manually, using N chunks of code, each with a one-line difference: `filter(facet_var == val1)` ... `filter(facet_var == valN)` 

In short, it seems to me that facetting + highlighting would be way more informative if it enabled within-group comparisons.

**Comment by wilfredom** [comment]
Examples: the non-highlighted geoms in the facetted charts should match the filtered ones:
![screen shot 2018-11-29 at 11 44 56 am](https://user-images.githubusercontent.com/15095683/49247602-8aa8db00-f3cc-11e8-9156-cb4694b49ae7.png)

**Comment by wilfredom** [comment]
Compare to the two below, you can see that the non-highlighted peaks are the same in the facetted version, but clearly not the same in the filtered versions:

![screen shot 2018-11-29 at 11 45 51 am](https://user-images.githubusercontent.com/15095683/49247743-dfe4ec80-f3cc-11e8-98c0-f2dfad10e14f.png)
![screen shot 2018-11-29 at 11 45 45 am](https://user-images.githubusercontent.com/15095683/49247744-dfe4ec80-f3cc-11e8-8873-6f81a8557019.png)

**Comment by yutannihilation** [comment]
Thanks, I didn't notice the need for the option to control this. I got it!

**Comment by wilfredom** [comment]
You rock!

**Comment by yutannihilation** [comment]
`use_facet_variables = TRUE` to add additional grouping variables.

**Comment by wilfredom** [comment]
Looking forward to trying this out; amazing!

**Comment by yutannihilation** [comment]
Ah, `use_facet_variables` is still just an idea... Sorry for addressing this late. Stay tuned!

**Comment by wilfredom** [comment]
Haha, no prob—thanks for the note :).

**Comment by yutannihilation** [comment]
@wilfredom I implemented `use_facet_vars` option. Could you confirm this enables what you want to do?

``` r
library(gghighlight)
#> Loading required package: ggplot2
library(patchwork)

d <- data.frame(
  idx =   c(1, 2, 3, 4, 1, 2, 3, 4),
  value = c(10, 11, 12, 13, 4, 8, 16, 32),
  cat1 =  rep(c("a", "b"), each = 4),
  cat2 =  rep(rep(c("1-2", "3-4"), each = 2), 2),
  stringsAsFactors = FALSE
)

p <- ggplot(d, aes(idx, value, colour = cat1)) +
  geom_line() +
  facet_wrap(vars(cat2))

p1 <- p +
  gghighlight(max(value) > 10) +
  ggtitle("(normal)")
#> label_key: cat1

p2 <- p +
  gghighlight(max(value) > 10, use_facet_vars = TRUE) +
  ggtitle("use_facet_vars = TRUE")
#> label_key: cat1

p1 / p2
```

![](https://i.imgur.com/nakeBwt.png)

<sup>Created on 2019-03-17 by the [reprex package](https://reprex.tidyverse.org) (v0.2.1)</sup>
